src/pres/loader/PGNLoader.js

```diff
@@ -16,36 +16,39 @@ export default class PGNLoader extends React.Component {
     constructor(props) {
         super(props)
         let selectedSite = new URLSearchParams(window.location.search).get("source")
+        let playerName = new URLSearchParams(window.location.search).get("playerName")
+        let color = new URLSearchParams(window.location.search).get("color") // 'white' | 'black'
+        // 'all' | 'rated' | 'casual' (corresponds to Filters.toggleRated)
+        let ratedMode = new URLSearchParams(window.location.search).get("ratedMode")
+        let fromDate = new URLSearchParams(window.location.search).get("fromDate") // YYYY-MM-DD
+        let toDate = new URLSearchParams(window.location.search).get("toDate") // YYYY-MM-DD
+        let timeControls = this.getTimeControls()
 
         this.state = {
-            playerName: '',
+            playerName: playerName?playerName:'',
             site: selectedSite?selectedSite:'',
-            playerColor: this.props.settings.playerColor,
+            playerColor: color ? color : this.props.settings.playerColor,
             isAdvancedFiltersOpen: false,
             isGamesSubsectionOpen: false,
-            expandedPanel: selectedSite?'user':'source',
+            expandedPanel: this.getInitiallyExpandedPanel(selectedSite, playerName),
             notablePlayers:null,
             notableEvents:null,
             files:[],
             selectedNotableEvent:{},
             selectedNotablePlayer:{},
             lichessLoginState: Constants.LICHESS_NOT_LOGGED_IN,
-            lichessLoginName: null
+            lichessLoginName: null,
+            ...timeControls
         }
         if(selectedSite === Constants.SITE_LICHESS) {
             this.fetchLichessLoginStatus()
         }
         this.state[Constants.FILTER_NAME_DOWNLOAD_LIMIT] = Constants.MAX_DOWNLOAD_LIMIT
-        this.state[Constants.TIME_CONTROL_ULTRA_BULLET] = true
-        this.state[Constants.TIME_CONTROL_BULLET] = true
-        this.state[Constants.TIME_CONTROL_BLITZ] = true
-        this.state[Constants.TIME_CONTROL_RAPID] = true
-        this.state[Constants.TIME_CONTROL_CLASSICAL] = true
-        this.state[Constants.TIME_CONTROL_CORRESPONDENCE] = true
-        this.state[Constants.TIME_CONTROL_DAILY] = true
-        this.state[Constants.FILTER_NAME_RATED] = "all"
+        this.state[Constants.FILTER_NAME_RATED] = ratedMode ? ratedMode : "all"
         this.state[Constants.FILTER_NAME_ELO_RANGE] = [0, Constants.MAX_ELO_RATING]
         this.state[Constants.FILTER_NAME_OPPONENT] = ''
+        this.state[Constants.FILTER_NAME_FROM_DATE] = fromDate ? parseDate(fromDate) : null
+        this.state[Constants.FILTER_NAME_TO_DATE] = toDate ? parseDate(toDate) : null
```

Comment:
These date state vars used to be `undefined` when empty, but I used `null` because it looks more in line with what you've got in the rest of your codebase. Not sure if changing here might have any unintended consequences